src/app/shared/services/course.service.ts

```diff
@@ -20,11 +21,11 @@ export class CourseService {
 
   onFetchCourses() {
     this.loadingService.loadingSubject.next(true);
-    this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?start=0&count=${this.quanityOfVisibleCourses}`)
-      .subscribe(response => {
-        this.coursesSubject.next(response);
-      })
-    this.loadingService.loadingSubject.next(false);
+    let obs = this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?start=0&count=${this.quanityOfVisibleCourses}`)
+    if (obs) {
+      this.loadingService.loadingSubject.next(false);
```

Comment:
Try to refactor it using RxJs

Reply:
ok, I have included changing of loadingSubject via pipe here.